src/Assignments/SimpleProblemPage.tsx

```diff
@@ -89,39 +89,67 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
+                    if (!_.isNil(currentTopic.topicAssessmentInfo.maxVersions)) {
+                        currentVersionsRemaining = currentTopic.topicAssessmentInfo.maxVersions - currentTopic.topicAssessmentInfo.studentTopicAssessmentInfo.length;
+                        setVersionsRemaining(currentVersionsRemaining);
+                    }
+
+                    // if the assessment has been "closed" OR time has expired - do NOT allow submissions...
+                    if ((!_.isNil(currentVersion.isClosed) && currentVersion.isClosed) ||
+                        (!_.isNil(currentVersion.endTime) && currentVersion.endTime.toMoment().isBefore(moment()))
+                    ) {
+                        if (attemptsRemaining !== 0) setAttemptsRemaining(0); // avoid render loop when exam ends without having used all available attempts
+                        if (currentTopic.topicAssessmentInfo.hideProblemsAfterFinish) {
+                            if (currentVersionsRemaining > 0) {
+                                setConfirmationParameters({
+                                    show: true,
+                                    onHide: () => setConfirmationParameters(DEFAULT_CONFIRMATION_PARAMETERS),
+                                    onConfirm: () => confirmStartNewVersion(currentTopic, currentVersionsRemaining),
+                                    headerContent: 'This version has expired',
+                                    bodyContent: 'Would you like to start a new version of this assessment?'
+                                });
+                            } else {
+                                setConfirmationParameters({
+                                    show: true,
+                                    onHide: () => setConfirmationParameters(DEFAULT_CONFIRMATION_PARAMETERS),
+                                    onConfirm: () => { },
+                                    headerContent: 'This version has expired',
+                                    bodyContent: 'You have completed all available versions of this assessment.'
+                                });
+                            }
+                        }
                     } else {
-                        setAttemptsRemaining(0);
+                        setAttemptsRemaining(currentAttemptsRemaining);
                     }
-                }
-                if (!_.isNil(topic.topicAssessmentInfo.maxVersions)) {
-                    setVersionsRemaining(topic.topicAssessmentInfo.maxVersions - topic.topicAssessmentInfo.studentTopicAssessmentInfo.length);
+                } else {
+                    console.error('Something is wrong with this assessment version. There are versions provided, but no version is "current"');
                 }
                 if (!_.isNil(currentVersion) && !_.isNil(currentVersion.id)) {
                     setVersionId(currentVersion.id);
                 }
-                setTopic(topic);
+                // setTopic(topic);
```

Comment:
Why are we commenting this out?

Reply:
setTopic is already handled above -- this was an artifact that didn't get cleaned up